kodilla-stream/src/main/java/com/kodilla/optional/homework/Applocation.java

```diff
@@ -0,0 +1,22 @@
+package com.kodilla.optional.homework;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class Applocation {
+    public static void main(String[] args) {
+        List<Student> students = new ArrayList<>();
+        students.add(new Student("Piotr", new Teacher("Lucyna Migdał")));
+        students.add(new Student("Agnieszka", new Teacher("Roman Ząbek")));
+        students.add(new Student("Krzysztof", new Teacher("null")));
+        students.add(new Student("Monika", new Teacher("null")));
+
+        for (Student student : students) {
+            Teacher teacher = student.getTeacher();
```

Comment:
Zamiast linii 16 -- 19 trzeba tak:
System.out.println("Student: " + student.getName() + ", nauczyciel: " + Optional.ofNullable(student.getTeacher()).map(teacher -> teacher.getName()).orElse("<undefined>"));

Reply:
WoW! a to tak można?! No szok! Szczęka mi opadła...

Reply:
nie wpadłabym na taki pomysł, by w sout wpisywać operacje:
1. myślałam. że Typ Optional stosuje się poza printem
2. jeszcze dałeś operację transformującą.
No szok po prostu. Świetne to rozwiązanie bo zastępuje moje 4ry linijki :)